src/transient_model.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+from build_matrix_A import build_matrix_A
+from build_matrix_b import build_matrix_b
+from matrix_solver import solve_matrix
+from generate_mesh import generate_mesh
+
+def transient_model(time, material_property_library, mesh, g_dot_list, boundary_conditions_list):
+    """This function will calculate the transient response of the fuel particle to changes in either heat
+    generation rate or to boundary wall temperature. The A matrix has 1 added to its diagonal entries while
+    the b matrix has the previous temperature set added to its entries. These changes need to be made now that 
+    dT/dt is no longer zero:
+    
+    T_time+1 = T_time + dT/dt *Delta_t
+    
+    Where previously dT/dt was equal to zero, the matrices simplified dramatically. Now this is not the case,
+    so the b matrix must be appended with the known constant T_time and the A matrix must be appended with 1 
+    along the diagonal to account for the presence of T_time+1.
+       
+    inputs
+    ------
+    - time: An array of the time steps taken by the solver. Passed through to the matrix generators.
+    - mesh: the array of points for the physical location of nodes and the materials assigned to them
+    - material_property_library: a dictionary or nested dictionaries of the materials used in the modeled region
+    - g_dot_list: a list with the heat generation rates before the transient and during the transient. g_dot_list 
+    is of length 2, with the 0th entry being prior to the transient and fist entry after time zero.
+    - boundary_condtions_list: a list with the boundary temperature before the transient and during the transient
+    outputs. This list is of length 2, with the 0th entry being prior to the transient and fist entry after time zero.
+    -T: numpy array containing the temperatures for each node and time
+    """  
+    
+    total_nodes = len(mesh[0][1])
+    for i in range(1,len(mesh)):
+        total_nodes += len(mesh[i][1])-1
+    T = np.zeros((total_nodes, len(time)))
+    Dt = time[1]-time[0]    
+    
+    A = Dt*build_matrix_A(material_property_library, mesh)
+    b = Dt*build_matrix_b(boundary_conditions_list[0], mesh, material_property_library, g_dot_list[0]) 
+    
+    T_initial = solve_matrix(A,b)
+    
+    A += np.eye(A.shape[0])
```

Comment:
I'm still not sure how the following few lines matter?

Reply:
Because the boundary condition is imposed on the final node such that the last entry of b is the boundary temperature and the last entry of A is 1, I don't want to tamper with them when using lines 47 or 42.

Reply:
My question was unclear and not well placed in the code.  I am not sure why you solve T once in line 40 and then update A and b, and then never use the updated b.  It seems the first solution of T should already have these modifications?

You previously solved for T twice in this rough section of the code, and I commented on that before.  Perhaps it was the second solve for T that was correct?

Reply:
I solve for an initial condition (steady state case) using the unaltered A and b matrices. You are correct that lines 47 and 48 are unnecessary, and are relics from the rougher version of the code where line 50 involved resolving the matrix equation.